crates/db_views/src/post_report_view.rs

```diff
@@ -24,26 +23,22 @@ use lemmy_db_schema::{
     post_report,
   },
   source::community::CommunityFollower,
-  utils::{
-    actions,
-    actions_alias,
-    functions::coalesce,
-    get_conn,
-    limit_and_offset,
-    DbConn,
-    DbPool,
-    ListFn,
-    Queries,
-    ReadFn,
-  },
+  utils::{actions, actions_alias, functions::coalesce, get_conn, DbPool},
 };
 
-fn queries<'a>() -> Queries<
-  impl ReadFn<'a, PostReportView, (PostReportId, PersonId)>,
-  impl ListFn<'a, PostReportView, (PostReportQuery, &'a LocalUserView)>,
-> {
-  let all_joins = |query: post_report::BoxedQuery<'a, Pg>, my_person_id: PersonId| {
-    query
+impl PostReportView {
+  /// returns the PostReportView for the provided report_id
+  ///
+  /// * `report_id` - the report id to obtain
+  pub async fn read(
+    pool: &mut DbPool<'_>,
+    report_id: PostReportId,
+    my_person_id: PersonId,
+  ) -> Result<Self, Error> {
```

Comment:
Is this function really needed? At least lemmy-ui doesnt seem to use it. Would also be possible to make a single `/report/read` endpoint which takes `ReportCombinedId`